priam/src/main/java/com/netflix/priam/services/SnapshotMetaService.java

```diff
@@ -0,0 +1,227 @@
+            //processOldSnapshotV2Folders();
+
+            //Take a new snapshot
+            cassandraOperations.takeSnapshot(snapshotName);
+
+            //Process the snapshot and upload the meta file.
+            processSnapshot(snapshotInstant).uploadMetaFile(true);
+
+            logger.info("Finished processing snapshot meta service");
+        } catch (Exception e) {
+            logger.error("Error while executing SnapshotMetaService", e);
+        }
+
+    }
+
+    MetaFileWriterBuilder.UploadStep processSnapshot(Instant snapshotInstant) throws Exception {
+        dataStep = metaFileWriter.newBuilder().startMetaFileGeneration(snapshotInstant);
+        initiateBackup(SNAPSHOT_FOLDER, backupRestoreUtil);
+        return dataStep.endMetaFileGeneration();
+    }
+
+    private File getValidSnapshot(File snapshotDir, String snapshotName) {
+        for (File fileName : snapshotDir.listFiles())
+            if (fileName.exists() && fileName.isDirectory() && fileName.getName().matches(snapshotName))
+                return fileName;
+        return null;
+    }
+
+    @Override
+    public String getName() {
+        return JOBNAME;
+    }
+
+
+    private ColumnfamilyResult convertToColumnFamilyResult(String keyspace, String columnFamilyName, Map<String, List<FileUploadResult>> filePrefixToFileMap) {
+        ColumnfamilyResult columnfamilyResult = new ColumnfamilyResult(keyspace, columnFamilyName);
+        filePrefixToFileMap.entrySet().forEach(sstableEntry -> {
+            ColumnfamilyResult.SSTableResult ssTableResult = new ColumnfamilyResult.SSTableResult();
+            ssTableResult.setPrefix(sstableEntry.getKey());
+            ssTableResult.setSstableComponents(sstableEntry.getValue());
+            columnfamilyResult.addSstable(ssTableResult);
+        });
+        return columnfamilyResult;
+    }
+
+    @Override
+    protected void processColumnFamily(final String keyspace, final String columnFamily, final File backupDir) throws Exception {
+        File snapshotDir = getValidSnapshot(backupDir, snapshotName);
+        // Process this snapshot folder for the given columnFamily
+        if (snapshotDir == null) {
+            logger.warn("{} folder does not contain {} snapshots", backupDir, snapshotName);
+            return;
+        }
+
+        logger.debug("Scanning for all SSTables in: {}", snapshotDir.getAbsolutePath());
+
+        Map<String, List<FileUploadResult>> filePrefixToFileMap = new HashMap<>();
+        Collection<File> files = FileUtils.listFiles(snapshotDir, FileFilterUtils.fileFileFilter(), null);
+
+        files.stream().filter(file -> file.exists()).filter(file -> file.isFile()).forEach(file -> {
+            try {
+                String prefix = PrefixGenerator.getSSTFileBase(file.getName());
+
+                if (prefix == null && file.getName().equalsIgnoreCase(CASSANDRA_MANIFEST_FILE))
```

Comment:
The goal of SnapshotMetaService is to capture filesystem at a moment in time. It will though "try" to push those files for upload via BackupUploadDownloadService (in next version). It would be completely ok to fail to enqueue the files to upload as there will be another service pushing all the files in the queue for upload (think of this like a cleanup thread and will help us in "resuming" any failed backup for any reason).